packages/react-server/src/render/render.tsx

```diff
@@ -85,7 +85,12 @@ export function createRender(render: RenderFunction, options: Options = {}) {
           <HydrationContext.Provider value={hydrationManager}>
             <NetworkContext.Provider value={networkManager}>
               {children}
-              <Serialize data={() => ctx.state.quiltData} />
+              <Serialize
+                data={() => ({
+                  ...ctx.state.quiltData,
+                  accessedHeaders: networkManager.accessedHeaders,
```

Comment:
I don't think we want to serialize this as part of the `data` object. I'd expect it to have it's own key.